GoogleForCreators/web-stories-wp
# Dashboard: File Upload UI Component

## Summary
Settings view in Dashboard needs a simple drag and drop file upload component. Drew on the editor for structure.

<img width="642" alt="Screen Shot 2020-07-08 at 1 10 45 PM" src="https://user-images.githubusercontent.com/10720454/86965504-8aca9a00-c11c-11ea-8237-e6cc34716de0.png">
<img width="621" alt="Screen Shot 2020-07-08 at 1 10 52 PM" src="https://user-images.githubusercontent.com/10720454/86965507-8bfbc700-c11c-11ea-932c-98030fba1c9a.png">
<img width="993" alt="Screen Shot 2020-07-09 at 11 24 59 AM" src="https://user-images.githubusercontent.com/10720454/87076870-2ddfea00-c1d7-11ea-9316-f8094ad393dc.png">


## Relevant Technical Choices
- Relies on native event handler functionality for drag and drop and file upload 
- Passes files to parent to process how the parent needs data to avoid locking use cases down 
- Delete functionality is passing the index of the upload as well as the file itself back to parent because of unknowns at the moment. 
- Uses story editor's `getResourceFromLocalFile` util to try and keep structures in sync behind the surface 

## To-do
- [x] Update icon from placeholder x to upload svg - need permissions in figma to download assets 
- [x] Determine 'delete' functionality 
- [ ] Follow up on when the file name should be blue like in figma 
- [ ] Follow up on if the "You can also drag your logo here" text should only be visible if there's not currently uploads (I think this is the case?) 
- [ ] Follow up on delete functionality (this will be after updated designs, read something about the ability to delete or also edit)
- [x] Get updated button style once PR for dialogs is merged

## User-facing changes
- Changes only exist in storybook 

## Testing Instructions
- See storybook Dashboard/Components/FileUpload 
- Upload file(s) with "button" (it's a label) and try drag and dropping, you should see an image show up for each with file name
- Hover over an uploaded image and click the 'x' to delete it, see actions trigger in storybook. 
- See storybook Dashboard/Views/EditorSettings to see the fileUpload in the settings page. Please note that the dummy functionality isn't hooked up in the view storybook ONLY the components since the view relies on context and since we don't know for sure what all is going to happen here re hooking up the real data I didn't want to spend the time on it. 

![drop upload](https://user-images.githubusercontent.com/10720454/86965678-c2394680-c11c-11ea-9302-fde1443b25c3.gif)
![file upload](https://user-images.githubusercontent.com/10720454/86965982-383dad80-c11d-11ea-8e59-3f8618df5b99.gif)

(placeholder UI for functionality) 
![placeholder delete](https://user-images.githubusercontent.com/10720454/86965672-c1081980-c11c-11ea-9589-1310c83e4d19.gif)

---



Addresses #2747
